wps_jobs/wps_jobs/fuel_raster.py

```diff
@@ -0,0 +1,144 @@
+    5. If validation succeeds, extracts raster dimensions and saves metadata to the database.
+
+    Args:
+        raster_addresser (RasterKeyAddresser): An instance used to generate S3 keys for raster objects.
+        start_datetime (datetime): The datetime associated with the raster job.
+        unprocessed_object_name (str): The S3 object name of the unprocessed raster.
+        expected_hash (str): The expected content hash of the raster file for validation.
+
+    Raises:
+        ValueError: If the content hash validation fails or the raster cannot be stored.
+    """
+    async with S3Client() as s3_client:
+        current_version = await find_latest_version(
+            s3_client, RasterKeyAddresser(), start_datetime, 1
+        )
+        unprocessed_key = raster_addresser.get_unprocessed_fuel_raster_key(unprocessed_object_name)
+        new_key = raster_addresser.get_fuel_raster_key(start_datetime, current_version + 1)
+
+        try:
+            await s3_client.copy_object(unprocessed_key, new_key)
+            res = await s3_client.get_fuel_raster(new_key, expected_hash)
+        except ValueError as e:
+            logger.error(f"Could not store fuel raster at: {new_key}", exc_info=e)
+            await s3_client.delete_object(new_key)
+            raise e
+
+        logger.info("Raster file content hash succeeded")
+        with WPSDataset.from_bytes(res) as new_raster_ds:
+            xsize = new_raster_ds.as_gdal_ds().RasterXSize
+            ysize = new_raster_ds.as_gdal_ds().RasterYSize
+            async with get_async_write_session_scope() as db_session:
+                now = get_utc_now()
+                await save_processed_fuel_raster(
+                    db_session,
+                    FuelTypeRaster(
+                        year=start_datetime.year,
+                        xsize=xsize,
+                        ysize=ysize,
+                        object_store_path=new_key,
+                        content_hash=expected_hash,
+                        create_timestamp=now,
+                    ),
+                )
+
+
+def main():
+    """Kicks off asynchronous processing of new fuel raster"""
+    parser = argparse.ArgumentParser(
+        description="Retrieve and store the latest fuel raster by date"
+    )
+    parser.add_argument(
+        "-d", "--date", default=None, help="The date to use for looking up the fuel raster."
+    )
+    parser.add_argument(
+        "-k", "--key", default=None, help="Object storage key that points to the unprocessed raster"
+    )
+    parser.add_argument(
+        "-e",
+        "--expected-hash",
+        default=None,
+        help="Expected content hash of the unprocessed raster's raw bytes",
+    )
+
+    args, _ = parser.parse_known_args()
+    start_datetime = datetime.fromisoformat(args.date) if args.date else get_utc_now()
+    unprocessed_object_name = str(args.key) if args.key else config.get("FUEL_RASTER_NAME")
+    expected_hash = (
```

Comment:
Why does the caller need to supply the expected hash? We use this job when adding a new fuel raster so I'm not sure why can't we just determine the has ourselves?

Reply:
Yeah this could be simplified, I'll remove the expected hash argument.